### PR TITLE
Provide a uniform api between Simple, KeyValue and Chain stores

### PR DESCRIPTION
**Pull request by kidpollo**
Provide a uniform api between Simple, KeyValue and Chain stores for public and private methods.

This is because a couple of gems I use out there need to call some of the internal methods of the simple store. When you try to use other gems configured with a Chain backend call those calls fail.

Basically I added the initialized?, translations and init_translations to the KeyValue and Chain stores.

Also added the functionality to be able to read back the translations from a key value store back to a hash as if it where read from the YAML files.

The gems that i know that would benefit from this would be i18n-js and babilu

### CODE REVIEW

**Comment by radar** [comment]
@kidpollo Yes I like this idea too. Could you please update your branch against the latest master? Will look at releasing this in 0.9.0.

**Comment by kidpollo** [comment]
Woah 5 years later still relevant?

**Comment by radar** [comment]
@kidpollo Do you think it would be relevant? Is it still useful to provide a uniform API?

**Comment by kidpollo** [comment]
@radar updated

**Comment by oniofchaos** [comment]
Still seems useful if you want to make another attempt at getting it cleaned up and merged.

**Comment by kidpollo** [comment]
Updated again

File: README.md [hunk 1]
```diff
@@ -115,6 +115,7 @@ Additional documentation can be found here: https://github.com/svenfuchs/i18n/wi
 * [Stephan Soller](http://www.arkanis-development.de)
 * [Saimon Moore](http://saimonmoore.net)
 * [Matt Aimonetti](https://matt.aimonetti.net/)
+* [Francisco Viramontes](https://twitter.com/kidpollo)
```
Comment (radar) [inline README.md]: Please remove this line. We do not add PR authors to this list. We have our list of contributors underneath this, but it's just a link to https://github.com/svenfuchs/i18n/graphs/contributors.

File: test/backend/chain_test.rb [hunk 2]
```diff
@@ -81,6 +81,32 @@ def setup
     I18n.backend.store_translations :foo, {:bar => :baz}, {:option => 'persists'}
   end
 
+  test 'store should call initialize on all backends and return true if all initialized' do
+    @first.send :init_translations
+    @second.send :init_translations
+    assert_equal(I18n.backend.initialized?, true)
```
Comment (radar) [inline test/backend/chain_test.rb]: I think these could be `assert I18n.backend.initialized?`

File: test/backend/chain_test.rb [hunk 3]
```diff
@@ -81,6 +81,32 @@ def setup
...
+    @first.send :init_translations
+    @second.send :init_translations
+    I18n.backend.reload!
+    assert_equal(@first.initialized?, false)
+    assert_equal(@second.initialized?, false)
+  end
+
+  test 'should be able to get all translations of the first backend' do
+    assert_equal I18n.backend.send(:translations),{:en => {
+      :foo=>"Foo", :formats=>{:short=>"short", :subformats=>{:short=>"short"}}, :plural_1=>{:one=>"%{count}"}, :dates=>{:a=>"A"}
```
Comment (radar) [inline test/backend/chain_test.rb]: Please split this hash out onto individual lines for each key, and indent each sub-key:

```ruby
{ 
  a: "Foo",
  formats: {
    short: "short",
    subformats:
```

File: test/backend/key_value_test.rb [hunk 4]
```diff
@@ -90,4 +103,5 @@ def assert_flattens(expected, nested, escape=true, subtree=true)
     assert_equal 'default', I18n.t(:missing_bar, count: 1, default: 'default')
     assert_equal 'default', I18n.t(:missing_bar, count: 0, default: 'default')
   end
+
```
Comment (radar) [inline test/backend/key_value_test.rb]: This extra blank line is not required.

**Comment by radar** [comment]
Thank you for keeping this continuously up to date. I have a few comments for you, and then I think this is good to merge :)

**Comment by kidpollo** [comment]
Done @radar

**Comment by radar** [comment]
Thank you!

I will merge this on Monday when I am back at work.

**Pull request merged**